userspace/libsinsp/plugin.cpp

```diff
@@ -645,6 +659,71 @@ std::string sinsp_plugin::str_from_alloc_charbuf(const char* charbuf)
 	return str;
 }
 
+bool sinsp_plugin::resolve_dylib_field_arg(Json::Value root, filtercheck_field_info &tf)
+{
+	if (root.isNull())
+	{
+		return false;
+	}
+
+	const Json::Value &isRequired = root.get("isRequired", Json::Value::null);
+	if (!isRequired.isNull())
+	{
+		if (!isRequired.isBool())
+		{
+			throw sinsp_exception(string("error in plugin ") + m_name + ": field " + tf.m_name + " isRequired property is not boolean");
+		}
+
+		if (isRequired.asBool() == true)
+		{
+			// All the extra casting is because this is the one flags value
+			// that is strongly typed and not just an int.
+			tf.m_flags = (filtercheck_field_flags) ((int) tf.m_flags | (int) filtercheck_field_flags::EPF_ARG_REQUIRED);
+		}
+	}
+
+	const Json::Value &isNumeric = root.get("isNumeric", Json::Value::null);
+	if (!isNumeric.isNull())
+	{
+		if (!isNumeric.isBool())
+		{
+			throw sinsp_exception(string("error in plugin ") + m_name + ": field " + tf.m_name + " isNumeric property is not boolean");
+		}
+
+		if (isNumeric.asBool() == true)
+		{
+			// We set `EPF_ARG_ALLOWED` implicitly.
+			tf.m_flags = (filtercheck_field_flags) ((int) tf.m_flags | (int) filtercheck_field_flags::EPF_ARG_NUMERIC);
+			tf.m_flags = (filtercheck_field_flags) ((int) tf.m_flags | (int) filtercheck_field_flags::EPF_ARG_ALLOWED);
+		}
+	}
+
+	const Json::Value &isString = root.get("isString", Json::Value::null);
+	if (!isString.isNull())
+	{
+		if (!isString.isBool())
+		{
+			throw sinsp_exception(string("error in plugin ") + m_name + ": field " + tf.m_name + " isString property is not boolean");
+		}
+
+		if (isString.asBool() == true)
+		{
+			// We set `EPF_ARG_ALLOWED` implicitly.
+			tf.m_flags = (filtercheck_field_flags) ((int) tf.m_flags | (int) filtercheck_field_flags::EPF_ARG_STRING);
+			tf.m_flags = (filtercheck_field_flags) ((int) tf.m_flags | (int) filtercheck_field_flags::EPF_ARG_ALLOWED);
+		}
+	}
+
+	if((tf.m_flags & filtercheck_field_flags::EPF_ARG_REQUIRED 
+			|| tf.m_flags & filtercheck_field_flags::EPF_ARG_ALLOWED) 
+		&& !(tf.m_flags & filtercheck_field_flags::EPF_ARG_NUMERIC 
+			|| tf.m_flags & filtercheck_field_flags::EPF_ARG_STRING))
+	{
+		throw sinsp_exception(string("error in plugin ") + m_name + ": field " + tf.m_name + " arg has isRequired true, but none of isString nor isNumeric is true");
```

Comment:
I am not sure about:
> 			|| tf.m_flags & filtercheck_field_flags::EPF_ARG_ALLOWED) 

Can it happen? Considering we set it implicitly.  
I mean, it clashes with the error message too (that only quotes 'isRequired').

Reply:
You're right that check is not necessary it was more for the sake of clarity. I agree with you that we have to remove it as it only decreases the performance evaluating the second `if` statement even when it is not necessary. I will remove that, thank you.